src/System.Linq/src/System/Linq/Single.cs

```diff
@@ -91,8 +93,16 @@ public static TSource SingleOrDefault<TSource>(this IEnumerable<TSource> source)
             {
                 switch (list.Count)
                 {
-                    case 0: return default(TSource);
-                    case 1: return list[0];
+                    case 0:
+                        {
+                            element = default(TSource);
+                            return TrySingleResult.NoElements;
+                        }
+                    case 1:
+                        {
+                            element = list[0];
+                            return TrySingleResult.SingleElement;
+                        }
```

Comment:
Nit: Do not indent/add brackets in case clauses

```diff
+                    case 1:
+                        element = list[0];
+                        return TrySingleResult.SingleElement;
```